src/darwin/Framework/CHIP/CHIPDeviceOverXPC.m

```diff
@@ -0,0 +1,237 @@
+                                                       __auto_type handleRetainer = handle;
+                                                       (void) handleRetainer;
+                                                   });
+                                               }];
+            } else {
+                dispatch_async(clientQueue, ^{
+                    CHIP_LOG_ERROR("Failed to obtain XPC connection to invoke command");
+                    completion(nil, [NSError errorWithDomain:CHIPErrorDomain code:CHIPErrorCodeGeneralError userInfo:nil]);
+                });
+            }
+        }];
+}
+
+- (void)subscribeAttributeWithEndpointId:(NSUInteger)endpointId
+                               clusterId:(NSUInteger)clusterId
+                             attributeId:(NSUInteger)attributeId
+                             minInterval:(NSUInteger)minInterval
+                             maxInterval:(NSUInteger)maxInterval
+                             clientQueue:(dispatch_queue_t)clientQueue
+                           reportHandler:(void (^)(NSDictionary<NSString *, id> * _Nullable value,
+                                             NSError * _Nullable error))reportHandler
+                 subscriptionEstablished:(void (^_Nullable)(void))subscriptionEstablishedHandler
+{
+    CHIP_LOG_DEBUG("Subscribing attribute ...");
+    [_xpcConnection getProxyHandleWithCompletion:^(
+        dispatch_queue_t _Nonnull queue, CHIPDeviceControllerXPCProxyHandle * _Nullable handle) {
+        if (handle) {
+            CHIP_LOG_DEBUG("Setup report handler");
+            [self.xpcConnection
+                registerReportHandlerWithController:self.controller
+                                             nodeId:self.nodeId
+                                            handler:^(id _Nullable value, NSError * _Nullable error) {
+                                                if (value && ![value isKindOfClass:[NSDictionary class]]) {
+                                                    CHIP_LOG_ERROR("Unsupported report format");
+                                                    return;
+                                                }
+                                                NSUInteger receivedEndpointId = [value[kCHIPEndpointIdKey] unsignedIntegerValue];
+                                                NSUInteger receivedClusterId = [value[kCHIPClusterIdKey] unsignedIntegerValue];
+                                                NSUInteger receivedAttributeId = [value[kCHIPAttributeIdKey] unsignedIntegerValue];
+                                                if (error
+                                                    || ((receivedEndpointId == endpointId || endpointId == 0xffff)
+                                                        && (receivedClusterId == clusterId || clusterId == 0xffffffff)
+                                                        && (receivedAttributeId == attributeId || attributeId == 0xffffffff))) {
```

Comment:
We should not be using those (undocumented) magic numbers for wildcards.  Needs a different setup.